### PR TITLE
[IRGen] Support additional single payload enum cases in layout strings

### PR DESCRIPTION
**Pull request by drexin**
rdar://105837101

Adds layout string support for single payload enums with simple (non-scattered) extra inhabitant patterns

### CODE REVIEW

**Comment by drexin** [comment]
@swift-ci smoke test

File: test/Interpreter/layout_string_witnesses_static.swift [hunk 1]
```diff
@@ -257,6 +257,36 @@ func testExistentialReference() {
 
 testExistentialReference()
 
+func testSinglePayloadAnyObjectEnum() {
```
Comment (drexin) [inline test/Interpreter/layout_string_witnesses_static.swift:260]: TODO: add test case for empty payload

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test macos

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test linux

**Comment by drexin** [comment]
@swift-ci smoke test linux

**Comment by drexin** [comment]
@swift-ci smoke test linux

**Comment by drexin** [comment]
@swift-ci smoke test
